node/core/dispute-coordinator/src/participation/queues/mod.rs

```diff
@@ -160,6 +160,21 @@ impl PartialEq for ParticipationRequest {
 }
 #[cfg(test)]
 impl Eq for ParticipationRequest {}
+#[cfg(test)]
+impl Clone for ParticipationRequest {
+	fn clone(&self) -> Self {
+		ParticipationRequest {
+			candidate_receipt: self.candidate_receipt.clone(),
+			candidate_hash: self.candidate_hash.clone(),
+			session: self.session,
+			_request_timer: None,
+		}
+	}
+
+	fn clone_from(&mut self, source: &Self) {
+		*self = source.clone();
+	}
```

Comment:
This seems unused?

Reply:
True. My understanding is that I need to provide this function as a part of the `Clone` implementation. Is there a better way to handle these unused but required parts of a type?

Reply:
Oh, I see it's a part of the `Clone` trait! I've never seen it before. But it doesn't seem to be a required method, so you can remove it. The doc for it is interesting:

> `a.clone_from(&b)` is equivalent to `a = b.clone()` in functionality, but can be overridden to reuse the resources of a to avoid unnecessary allocations.